src/core/src/op/bitwise_left_shift.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/bitwise_left_shift.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/elementwise_args.hpp"
+#include "openvino/reference/bitwise_left_shift.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+namespace left_shift {
+struct Evaluate : ov::element::NoAction<bool> {
+    using ov::element::NoAction<bool>::visit;
+
+    template <element::Type_t ET>
+    static result_type visit(const Tensor& in0, const Tensor& in1, Tensor& out) {
+        using T = typename element_type_traits<ET>::value_type;
+        reference::bitwise_left_shift(in0.data<const T>(),
+                                      in1.data<const T>(),
+                                      out.data<T>(),
+                                      in0.get_shape(),
+                                      in1.get_shape());
+        return true;
+    }
+};
+
+namespace {
+bool evaluate(TensorVector& outputs, const TensorVector& inputs) {
+    using namespace ov::element;
+    return IF_TYPE_OF(bitshift_evaluate,
```

Comment:
> Use `IF_TYPE_OF_CONVERT_TENSORS` to integrate with. 

As I can see this macro will convert inputs to f32 in case of unsupported precision, it shouldn't be applied to Bitwise operations.

> Why these precisions are selected is required at all in core?

I32 and I64 are the types produced by ShapeOf, so those types are raleted to the constant folding and shape of subgraphs. 
The u8 type can be seen with Constant and bitwise right shift with quantization related u4 packing, so evaluate could be useful to constantfold similar cases.
Currently I32 can be useful for sure. Can be reduced to i32 only, if those additional two makes difference.

Reply:
(Reduced to i32 only)